okhttp/src/nonJvmMain/kotlin/okhttp3/Call.kt

```diff
@@ -16,9 +16,38 @@
  */
 package okhttp3
 
+import okio.IOException
+
 actual interface Call {
   actual fun request(): Request
 
+  /**
+   * Invokes the request immediately, and blocks until the response can be processed or is in error.
+   *
+   * To avoid leaking resources callers should close the [Response] which in turn will close the
+   * underlying [ResponseBody].
+   *
+   * ```kotlin
+   * // ensure the response (and underlying response body) is closed
+   * client.newCall(request).executeAsync().use {
+   *   ...
+   * }
+   * ```
+   *
+   * The caller may read the response body with the response's [Response.body] method. To avoid
+   * leaking resources callers must [close the response body][ResponseBody] or the response.
+   *
+   * Note that transport-layer success (receiving a HTTP response code, headers and body) does not
+   * necessarily indicate application-layer success: `response` may still indicate an unhappy HTTP
+   * response code like 404 or 500.
+   *
+   * @throws IOException if the request could not be executed due to cancellation, a connectivity
+   *     problem or timeout. Because networks can fail during an exchange, it is possible that the
+   *     remote server accepted the request before the failure.
+   * @throws IllegalStateException when the call has already been executed.
+   */
+  suspend fun executeAsync(): Response
```

Comment:
Should we have blocking execute() on native?

Reply:
I think we need some concrete example clients to try this with, before adding more API.

There are different sweet spots that I think we will find get used

1) Multiplatform clients wanting a common base, client code in commonMain.
2) Individual platform specific calls even in overall multiplatform projects, client code in jvmMain, jsMain but maybe some common Interceptors.